Rubberduck.Parsing/Symbols/Declaration.cs

```diff
@@ -44,7 +44,7 @@ public class Declaration : IEquatable<Declaration>
             _parentScope = parentScope ?? string.Empty;
             _identifierName = qualifiedName.MemberName;
             _asTypeName = asTypeName;
-            _isSelfAssigned = isSelfAssigned;
+            _isSelfAssigned = isSelfAssigned || (declarationType == DeclarationType.Variable && parentDeclaration != null && parentDeclaration.IdentifierName == ComponentName);
```

Comment:
Why is determining this the Declaration's responsibility?

I'd have assumed that's what the creator of the object already knows...